### ISSUE TITLE
Generate API docs for the native (Android, iOS) parts of the services/plugins framework

### ISSUE THREAD

**Issue by qchong** [flutter/flutter#4546]
We should provide API reference docs for the classes and methods related to using `FlutterView`, for both iOS and Android developers.

Specifically, make sure to cover FlutterView (Android) and FlutterViewController (iOS) classes.

- For Java, suggest using [Javadoc](http://www.oracle.com/technetwork/java/javase/documentation/javadoc-137458.html) or the [Gradle Java Plugin](https://docs.gradle.org/current/userguide/java_plugin.html) to generate these HTML API docs.
- For Objective-C, suggest using [Doxygen](http://www.stack.nl/~dimitri/doxygen/) to generate these HTML API docs.

**Comment by eseidelGoogle** [comment]
I looked briefly at https://github.com/tomaz/appledoc just now, it appears to require that we have an Xcode project file (which we don't have for our FlutterView bits for example):
https://github.com/flutter/engine/tree/master/sky/shell/platform/ios/framework

**Comment by eseidelGoogle** [comment]
`javadoc` worked fine. :)  Just cd into https://github.com/flutter/engine/tree/master/sky/shell/platform/android and then run `javadoc io.flutter.view` and bam! :)

I'm not sure how we'll wire this into https://github.com/flutter/flutter/blob/master/dev/bots/docs.sh however since that's run from the flutter/flutter repository not flutter/engine.  Presumably we should run the docs as part of the flutter/engine build process, upload those to GCS and then let the flutter/flutter docs.sh download and upload them to firebase?

**Comment by eseidelGoogle** [comment]
I suspect this is going to need to bump up in priority, since drawing something other than pixels is a common thing for Apps to need to do. :)

**Comment by eseidelGoogle** [comment]
Looks like https://github.com/realm/jazzy may be the new hotness for iOS docs?

**Comment by eseidelGoogle** [comment]
Sounds like @jason-simmons may be interested in the java side, and @cbracken in the Obj-C side.

**Comment by mit-mit** [comment]
Assigning @jason-simmons as he seems to be fixing this already wrt. java.

**Comment by mit-mit** [comment]
@cbracken can I assign you for obj-c? And will the solution being considered there also work if we switch to Swift?

**Comment by cbracken** [comment]
I've got a few issues to crank through before I get to this, so if there's someone ready to go on this, they should feel free to. Otherwise, I'm happy to pick it up once I've got a couple of the iOS issues I'm working on off my plate.

A few options:
* [Doxygen](http://www.stack.nl/~dimitri/doxygen/): pretty standard doc generation tool. Doesn't yet have Swift support. Cross-platform.
* [Jazzy](https://github.com/realm/jazzy): ObjC/Swift focused doc-gen. Written in Ruby, so assuming it should run on Linux, but haven't tested.
* [AppleDoc](https://github.com/tomaz/appledoc): Obj-C only. No Swift support. No Linux version.
